src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -358,28 +358,38 @@ void distributeFlushTasks() {
                         logger.logDebug(
                             "buildUploadWorkers stats={}", this.buildUploadWorkers.toString());
                         return buildAndUpload(filePath, blobData);
-                      } catch (IOException e) {
+                      } catch (IOException
+                          | InvalidAlgorithmParameterException
+                          | NoSuchPaddingException
+                          | IllegalBlockSizeException
+                          | BadPaddingException
+                          | InvalidKeyException
+                          | NoSuchAlgorithmException e) {
+                        StringBuilder stackTrace = new StringBuilder();
+                        for (StackTraceElement element : e.getStackTrace()) {
```

Comment:
Nit, you could reduce to `e.getStackTrace().forEach(stackTrace.append(StackTraceElement::toString)` or something along those lines.

Reply:
getStackTrace() actually returns an array instead of a list